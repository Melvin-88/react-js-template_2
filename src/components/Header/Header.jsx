import React from 'react';

export default function HomeHeader(props) {

    return (
        <div className='Header'>
            <div className="logo">
                LOGO
            </div>
            <div className="user_select">

            </div>
        </div>
    )
}