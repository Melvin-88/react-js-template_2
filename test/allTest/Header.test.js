import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from '../../src/components/Header/Header';

let InitialData = {
    items: [1,2,3,4]
};

it('should render a document title and a parent title', () => {
    const wrapper = shallow(
        <Header items={InitialData} />
    );
    expect(wrapper).toMatchSnapshot();
    // Проверка на соответствие дитей в найденом елементе
    expect(wrapper.find('.logo').contains(<span>LOGO</span>)).toEqual(true);
    // Проверка на текст
    expect(wrapper.find('.logo').children().first().text()).toEqual("LOGO");
    // Проверка на  тип дочернего елемента
    expect(wrapper.find('.logo').childAt(0).type()).toEqual('span');
    // Проверка на существование
    expect(wrapper.find('.user_select').exists()).toEqual(true);
    // Проверка на пустоту
    expect(wrapper.find('.user_select').isEmpty()).toEqual(false);
    // Проверка изображения на урл ==
    expect(wrapper.find('.img_box').children().first().prop('src')).toEqual('111.png');
    // Указать стейт
    wrapper.setState({ url: '222.png' });
    // Еще одна проверка с указаной стейт
    expect(wrapper.find('.img_box').children().first().prop('src')).toEqual('222.png');
    // Проверка на тип дитей
    expect(wrapper.find('ul').childAt(0).type()).toEqual('li');
    // Проверка на количество дитей
    expect(wrapper.find('ul').children().length).toBe(InitialData.items.length);


    // expect(wrapper.find('img').prop('src').isEmpty()).toEqual(false);
});