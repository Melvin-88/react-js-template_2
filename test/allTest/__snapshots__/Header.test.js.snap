// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a document title and a parent title 1`] = `
<div
  className="Header"
>
  <div
    className="logo"
  >
    <span>
      LOGO
    </span>
  </div>
  <div
    className="user_select"
  >
    00:00
  </div>
  <div
    className="img_box"
  >
    <img
      alt=""
      src="111.png"
    />
  </div>
  <ul>
    <li
      key="0"
    >
      1
    </li>
    <li
      key="1"
    >
      2
    </li>
    <li
      key="2"
    >
      3
    </li>
    <li
      key="3"
    >
      4
    </li>
  </ul>
</div>
`;
